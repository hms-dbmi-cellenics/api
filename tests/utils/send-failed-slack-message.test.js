const fetchMock = require('jest-fetch-mock');
const sendFailedSlackMessage = require('../../src/utils/sendFailedSlackMessage');
const { USER } = require('../test-utils/constants');

fetchMock.enableFetchMocks();

jest.mock('../../src/utils/crypt', () => ({
  getWebhookUrl: () => 'http://mockSlackHook.com/asdasd',
}));
const experiment = {
  meta: {
    pipeline: {
      stateMachineArn: 'qcArn',
    },
    gem2s: {
      stateMachineArn: 'gem2sArn',
    },
  },
};

describe('sendFailedSlackMessage', () => {
  beforeEach(() => {
    fetch.mockClear();
    fetchMock.mockResponse(JSON.stringify({ ok: true }));
  });

  it('Sends slack message with correct fields for QC', async () => {
    const message = {
      input: {
        processName: 'qc',
        taskName: 'Task 1',
        authJWT: 'lmao',
      },
      experimentId: 'asd123',
    };

    await sendFailedSlackMessage(message, USER, experiment);
    expect(fetch.mock.calls).toMatchSnapshot();
  });

  it('Sends slack message with correct fields for Gem2s', async () => {
    const message = {
      input: {
        processName: 'gem2s',
        taskName: 'someStep',
        authJWT: 'lmao',
      },
      experimentId: 'anotherExpId',
    };
    await sendFailedSlackMessage(message, USER, experiment);
    expect(fetch.mock.calls).toMatchSnapshot();
  });
});
