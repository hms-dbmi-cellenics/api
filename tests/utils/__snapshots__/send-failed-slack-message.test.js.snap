// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendFailedSlackMessage Sends slack message with correct fields for Gem2s 1`] = `
Array [
  Array [
    "http://mockSlackHook.com/asdasd",
    Object {
      "body": "{\\"blocks\\":[{\\"type\\":\\"section\\",\\"text\\":{\\"type\\":\\"plain_text\\",\\"text\\":\\"\\\\n          The process for gem2s has failed for experiment anotherExpId \\\\n          Step: someStep \\\\n          State machine arn: gem2sArn \\\\n          \\"}},{\\"type\\":\\"context\\",\\"elements\\":[{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User email:*\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"johncena@gmail.com\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User name:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"John Cena\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User UUID:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"032cdb44-0cd3-4d58-af21-850kp0b95ac7\\"}]}]}",
      "method": "POST",
    },
  ],
]
`;

exports[`sendFailedSlackMessage Sends slack message with correct fields for QC 1`] = `
Array [
  Array [
    "http://mockSlackHook.com/asdasd",
    Object {
      "body": "{\\"blocks\\":[{\\"type\\":\\"section\\",\\"text\\":{\\"type\\":\\"plain_text\\",\\"text\\":\\"\\\\n          The process for qc has failed for experiment asd123 \\\\n          Step: Task 1 \\\\n          State machine arn: qcArn \\\\n          \\"}},{\\"type\\":\\"context\\",\\"elements\\":[{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User email:*\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"johncena@gmail.com\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User name:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"John Cena\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User UUID:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"032cdb44-0cd3-4d58-af21-850kp0b95ac7\\"}]}]}",
      "method": "POST",
    },
  ],
]
`;
